Content.Shared/Nyanotrasen/Abilities/Psionics/SharedPsionicAbilitiesSystem.cs

```diff
@@ -49,6 +51,14 @@ private void OnShutdown(EntityUid uid, PsionicsDisabledComponent component, Comp
                 TogglePsionics(uid, true);
         }
 
+        private void OnMobStateChanged(EntityUid uid, PsionicComponent component, MobStateChangedEvent args)
+        {
+            if (args.NewMobState == MobState.Dead)
+                TogglePsionics(uid, false);
+            else
+                TogglePsionics(uid, true);
```

Comment:
there's some edge cases where this will allow people to use psionics while insulated